### PR TITLE
add in basic snapshot functionality

### PR DESCRIPTION
**Pull request by daniel-kg**
This adds a couple of methods to create snapshots and revert to them.

### CODE REVIEW

File: test/unit/test_vm.py [hunk 1]
```diff
@@ -333,3 +335,90 @@ def test_get_all_virtual_machines(
...
+            CreateSnapshot_ret_val, 'Creating snapshot test_snapshot on test_vm',
+            vm.timeout
+        )
+        vm._vm_object.CreateSnapshot.assert_called_once_with(
+            'test_snapshot', 'bananas', True, False)
+
+    @mock.patch('vcdriver.vm.wait_for_vcenter_task')
+    def test_revert_snapshot(self, wfvc_mock):
+        vm = VirtualMachine(name='test_vm')
+        vm._vm_object = mock.MagicMock()
```
Comment (foolswood) [inline test/unit/test_vm.py]: In case you haven't come across this: you can spec mocks so they error on things that don't exist on the object.

Comment (daniel-kg) [inline test/unit/test_vm.py]: :+1:

File: test/unit/test_vm.py [hunk 2]
```diff
@@ -15,7 +15,9 @@
 from vcdriver.vm import (
     VirtualMachine,
     virtual_machines,
-    get_all_virtual_machines
+    get_all_virtual_machines,
+    _search_snapshots_by_name,
+    _get_snapshot_by_name
 )
```
Comment (Lantero) [inline test/unit/test_vm.py]: Is there a reason why these two functions are not in the tests file, as they are not used anywhere else?

Comment (daniel-kg) [inline test/unit/test_vm.py]: They're called from the revert_to_snapshot method on the virtual machine

**Comment by daniel-kg** [comment]
I'll give writing an integration test a little crack right now.

File: test/unit/test_vm.py (hunk 2 above)
Comment (Lantero) [inline test/unit/test_vm.py]: Oh, my bad! :+1:

**Pull request merged**